packages/code-du-travail-frontend/src/conventions/ConventionTexte.js

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import getConfig from "next/config";
+import Sidebar from "./texte/Sidebar";
+import Content from "./texte/Content";
+import styled from "styled-components";
+import { theme } from "@cdt/ui/";
+
+const {
+  publicRuntimeConfig: { API_DILA2SQL_URL }
+} = getConfig();
+
+class ConventionTexte extends React.Component {
+  constructor(props) {
+    super(props);
+    if (props.preloadedTexte) {
+      const texte = props.preloadedTexte;
+      const rootNode = this.getRootNode(texte);
+      this.state = { rootNode, texte, loaded: true };
+    } else {
+      this.state = { loaded: false };
+    }
+  }
+
+  async componentDidMount() {
+    if (this.state.texte) {
+      return false;
+    }
+    const { id } = this.props;
+    const texte = await this.fetchTexte({ id });
+    const rootNode = this.getRootNode(texte);
+    this.setState({ texte, rootNode, loaded: true });
+  }
+
+  getRootNode(startNode) {
```

Comment:
I found the name misleading, I imagine the function go up the tree to find the root node of a node but this function  seems to go down the children

Reply:
I get what you're saying, could `getFirstNodeWithChildren` be better?

Reply:
totally !